src/core/reactor.cc

```diff
@@ -4190,13 +4190,19 @@ unsigned smp::adjust_max_networking_aio_io_control_blocks(unsigned network_iocbs
         };
 
         if (available_aio >= requested_aio_other + smp::count) { // at least one queue for each shard
+            unsigned smp_count_recommended = get_smp_count(network_iocbs);
             network_iocbs = (available_aio - requested_aio_other) / smp::count;
             seastar_logger.warn("Your system does not have enough AIO capacity for optimal network performance; reducing `max-networking-io-control-blocks'.");
             seastar_logger.warn("Resultant AIO control block usage:");
             seastar_logger.warn("");
             log_aiocbs(log_level::warn, storage_iocbs, preempt_iocbs, network_iocbs, reserve_iocbs);
             seastar_logger.warn("");
-            seastar_logger.warn("For optimal network performance, set /proc/sys/fs/aio-max-nr to at least {}.", aio_nr + requested_aio);
+            std::string advice = format("For optimal network performance, set /proc/sys/fs/aio-max-nr to at least {}", aio_nr + requested_aio);
+            if (smp_count_recommended > 0) {
+                advice.append(format(", or decrease the logical CPU count of the application to {}", smp_count_recommended));
+            }
+            advice.append(".");
+            seastar_logger.warn("{}", advice);
```

Comment:
nit, despite that warning messages are always enabled, we could avoid constructing a `std::string` using something like:

```c++
            seastar_logger.warn("For optimal network performance, set /proc/sys/fs/aio-max-nr to at least {}"
                    "{}"
                    ".",
                    aio_nr + requested_aio,
                    smp_count_recommended > 0
                        ? ", or decrease the logical CPU count of the application to " + std::to_string(smp_count_recommended)
                        : "");
```

Reply:
Good observation; I'd thought about this question myself. I have two counter-arguments for this specific proposal:
- If we do something like this, then we need to modify both branches.
- I find the proposed update, as-is, harder to read than what's in the patch (and pre-patch, on the other branch). String concatenation with `+` and `std::to_string` is awkard; I find it harder to read than format strings. (Of course, with this argument, I'm slightly questioning my own code as well, which uses `std::string::append()` for concatenation, not a grand format string...)

What if we replace
~~~
", or decrease the logical CPU count of the application to " + std::to_string(smp_count_recommended)
~~~
with
~~~
format(", or decrease the logical CPU count of the application to {}", smp_count_recommended)
~~~
?

(If that's what we end up doing, then I should probably rework the preexistent "append logic" on the exception-throwing branch, too.)

Reply:
Resolved in v2 -- this code was part of v1 patch 3, but in v2, I've dropped patch 3.